### ISSUE TITLE
SBOM with zero components:  No "Completed metrics update for project"

### ISSUE THREAD

**Issue by msymons** [DependencyTrack/dependency-track#1183]
### Current Behavior:
When uploading an SBOM that contains ZERO components,  Dependency-Track will correctly show that the project has zero components and zero vulnerabilities.

However,  the logs show that there are no log events for `MetricsUpdateTask` for the project until what looks like a scheduled cron almost an hour later.  The log event for `VulnerabilityAnalysisTask` completion seems to also be delayed for over an hour.

This is causing Jenkins Dependency-Track plugin that is using "synchronous publishing mode" to time-out,  resulting in broken builds.

### Steps to Reproduce:

To test things I created a simple maven project with a single dependency:

```
    <dependencies>
        <dependency>
            <groupId>junit</groupId>
            <artifactId>junit</artifactId>
            <version>4.13.2</version>
            <scope>test</scope>
        </dependency>
    </dependencies>
```

In Jenkins, I used goal:

`verify org.cyclonedx:cyclonedx-maven-plugin:2.5.2:makeBom`

..and set up dependency-track-plugin as normal.

The build fails, timing out after (default) 5 minutes.  Extending the timeout to as long as 45 minutes does not help.

Editing the project POM and changing the scope for junit to compile resulted in a passing build.

I also tested with cyclonedx-maven-plugin 2.0.3, 2.4.1, 2.5.0 and 2.5.1 but this made no difference.  However, using 2.0.3 meant that the SBOM format was 1.2 (compared with 1.3 for cyclonedx-maven-plugin 2.5.2).

### Expected Behavior:
The results for the analysis of the project with zero components should be available immediately,  as they were in Dependency-Track 3.8.0

### Environment:
  
  - Dependency-Track Version: 4.3.4
  - Distribution: [ Docker  ]
  - BOM Format & Version: CycloneDX schema 1.2 or 1.3
  - Database Server: [ PostgreSQL ]

### Additional Details:

Normal log:

```
10:18:52.227 INFO [BomUploadProcessingTask] Processing CycloneDX BOM uploaded to project: fbace342-0148-4ad4-8d25-a12569f3bf13
10:18:52.995 INFO [BomUploadProcessingTask] Processing CycloneDX dependency graph for project: fbace342-0148-4ad4-8d25-a12569f3bf13
10:18:53.113 INFO [BomUploadProcessingTask] Processed 22 components and 0 services uploaded to project fbace342-0148-4ad4-8d25-a12569f3bf13
10:18:53.182 INFO [InternalAnalysisTask] Starting internal analysis task
10:18:53.229 INFO [InternalAnalysisTask] Internal analysis complete
10:18:53.242 INFO [OssIndexAnalysisTask] Starting Sonatype OSS Index analysis task
10:18:54.367 INFO [OssIndexAnalysisTask] Analyzing 7 component(s)
10:18:54.367 INFO [OssIndexAnalysisTask] Sonatype OSS Index analysis complete
10:18:54.371 INFO [PolicyEngine] Evaluating 22 component(s) against applicable policies
10:18:54.682 INFO [PolicyEngine] Policy analysis complete
10:18:54.696 INFO [MetricsUpdateTask] Executing metrics update for project: fbace342-0148-4ad4-8d25-a12569f3bf13
10:18:55.584 INFO [MetricsUpdateTask] Completed metrics update for project: fbace342-0148-4ad4-8d25-a12569f3bf13
10:19:15.697 INFO [MetricsUpdateTask] Completed portfolio metrics update
```

Log for 0 component BOM:

```
19:44:04.020 INFO [BomUploadProcessingTask] Processing CycloneDX BOM uploaded to project: 01f0615b-6067-4bc5-ba97-c4d8271f206e
19:44:04.093 INFO [BomUploadProcessingTask] Processing CycloneDX dependency graph for project: 01f0615b-6067-4bc5-ba97-c4d8271f206e
19:44:04.116 INFO [BomUploadProcessingTask] Processed 0 components and 0 services uploaded to project 01f0615b-6067-4bc5-ba97-c4d8271f206e
19:44:04.117 INFO [VulnerabilityAnalysisTask] Analyzing portfolio
19:44:04.173 INFO [VulnerabilityAnalysisTask] Analyzing 626 components in project: 0d73f72d-0e56-4b69-97bb-ff1e76f10677
19:44:04.192 INFO [InternalAnalysisTask] Starting internal analysis task
19:44:05.456 INFO [InternalAnalysisTask] Internal analysis complete
19:44:05.458 INFO [NpmAuditAnalysisTask] Starting Node Audit analysis task
19:44:06.231 INFO [MetricsUpdateTask] Completed metrics update for project: b5d50283-4791-434e-b938-7de78353f359
...
20:54:58.649 INFO [VulnerabilityAnalysisTask] Completed analysis of 0 components in project: 01f0615b-6067-4bc5-ba97-c4d8271f206e
```
The "Completed analysis of 0 components" after 70 mins makes would explain why my 45 minute timeout configuration was insufficient.

This issue is logged as a follow-on to an issues previously reported against Jenkins Dependency-Track Plugin [here](https://github.com/jenkinsci/dependency-track-plugin/issues/47).

Note that the author did not see the same problem...  but observed that they were using H2 database (compared with my PostgreSQL).

**Comment by jonbrohauge** [comment]
This bug is causing quite a headache for our developers. Is there an ETA for this being release? Quietly asking to see if it is necessary for us to do a temporary hack to disable'ish our Dependency-Track step in the build pipeline scripts without all the developers having to patch their pipelines twice.

**Comment by stevespringett** [comment]
Fixed in v4.3.6
